docs/user/file-size.md

```diff
@@ -20,7 +20,8 @@ writer = PdfWriter()
 for page in reader.pages:
     writer.add_page(page)
 
-writer.add_metadata(reader.metadata)
+if reader.metadata is not None:
+    writer.add_metadata(reader.metadata)
```

Comment:
Would it be better to enhance `add_metadata`? It could handle the special case that it was called with None.

Reply:
The type hints explicitly state this this should be a dictionary, thus at least for now I consider it a valid case to not support `None` there: https://pypdf.readthedocs.io/en/stable/modules/PdfWriter.html#pypdf.PdfWriter.add_metadata

Reply:
Okay, then this pull request is required to avoid an unsupported function call.